### PR TITLE
Use build submodule and adapt CI pipeline

### PR DESCRIPTION
**Pull request by turkenh**
### Description of your changes

This PR uses upbound/build submodule to build and publish configuration package (specifically https://github.com/upbound/build/pull/203 and https://github.com/upbound/build/pull/207).
It also adapts CI pipeline accordingly.

I have:

- [x] Read and followed Upbound's [contribution process](https://git.io/fj2m9).

### How has this code been tested

```
make build
```

Ensure package is build under  `_output/xpkg`
CI needs to be tested after merge.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -305,7 +305,7 @@ docker login ${REGISTRY} -u ${UPBOUND_ACCOUNT_EMAIL}
 Build package.
 
 ```console
-up xpkg build --name package.xpkg --ignore ".github/workflows/*,hack/*"
+up xpkg build --name package.xpkg --package-root=package
```
Comment (ytsarev) [inline README.md:308]: I like the direction but we were also implicitly including `examples` before, so maybe we can do
```suggestion
up xpkg build --name package.xpkg --package-root=package --examples-root="examples"
```

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+CROSSPLANE_NAMESPACE = upbound-system
+-include build/makelib/local.xpkg.mk
+
+# ====================================================================================
+# Targets
+
+# We must ensure up is installed in tool cache prior to build as including the k8s_tools machinery prior to the xpkg
+# machinery sets UP to point to tool cache.
+build.init: $(UP)
```
Comment (ytsarev) [inline Makefile:32]: can we add `submodules` target to the Makefile to enable the local `build` invocation?

**Review by ytsarev — COMMENTED** [review]
Overall looks great, left a couple of comments

File: README.md (hunk 1 above)
Comment (turkenh) [reply README.md]: Done!

File: Makefile (hunk 2 above)
Comment (turkenh) [reply Makefile:48]: Done!

**Review by ytsarev — APPROVED** [review]
Looks great 👍

**Review by jbw976 — APPROVED** [review]
LGTM @turkenh, thanks for adding this! from your PR notes, it looks like the CI integration will be tested after merge.  Anything else you think we can do to verify before merging? If not, I think you have permissions to merge yourself now :)

**Comment by turkenh** [comment]
We will need to add the following repo secrets before merge: XPKG_ACCESS_ID and XPKG_TOKEN

**Pull request merged**